x-pack/solutions/observability/packages/kbn-evals-suite-obs-ai-assistant/evals/knowledge_base/evaluate_knowledge_base_dataset.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Example } from '@arizeai/phoenix-client/dist/esm/types/datasets';
+import type { DefaultEvaluators, KibanaPhoenixClient } from '@kbn/evals';
+import type { EvaluationDataset } from '@kbn/evals/src/types';
+import type { ObservabilityAIAssistantEvaluationChatClient } from '../../src/chat_client';
+import type { KnowledgeBaseClient } from '../../src/knowledge_base_client';
+
+interface KnowledgeBaseExample extends Example {
+  input: {
+    question: string;
+  };
+  output: {
+    criteria?: string[];
+  };
+}
+
+export type EvaluateKnowledgeBaseDataset = ({
+  dataset: { name, description, examples },
+}: {
+  dataset: {
+    name: string;
+    description: string;
+    examples: KnowledgeBaseExample[];
+  };
+}) => Promise<void>;
+
+export function createEvaluateKnowledgeBaseDataset({
+  evaluators,
+  phoenixClient,
+  chatClient,
+  knowledgeBaseClient,
+}: {
+  evaluators: DefaultEvaluators;
+  phoenixClient: KibanaPhoenixClient;
+  chatClient: ObservabilityAIAssistantEvaluationChatClient;
+  knowledgeBaseClient: KnowledgeBaseClient;
+}): EvaluateKnowledgeBaseDataset {
+  return async function evaluateKnowledgeBaseDataset({
+    dataset: { name, description, examples },
+  }: {
+    dataset: {
+      name: string;
+      description: string;
+      examples: KnowledgeBaseExample[];
+    };
+  }) {
+    await knowledgeBaseClient.ensureInstalled().catch((e) => {
+      throw new Error(`Failed to install knowledge base: ${e.message}`);
+    });
+
+    const dataset = {
+      name,
+      description,
+      examples,
+    } satisfies EvaluationDataset;
```

Comment:
Any reason this is not sufficient?

```suggestion
    const dataset: EvaluationDataset = {
      name,
      description,
      examples,
    };
```

Reply:
I followed Dario's example in the ES|QL scenarios.
Having `satisfies` validates the shape without widening, so literals stay narrow.
For example:

If I remove `satisfies` type errors are thrown

<img width="1270" height="776" alt="image" src="https://github.com/user-attachments/assets/fe0a2183-2002-43a3-a260-85e1258db031" />

It happens because in `evaluate_knowledge_base_dataset.ts` where `expected.criteria` is accessed, `expected` widens to `Example['output']` which doesn’t have `criteria`.

Using `satisfies` here keeps `TEvaluationDataset` as the concrete type (with `KnowledgeBaseExample[]`), so `expected` is `KnowledgeBaseExample['output']` and `criteria` is correctly typed.